src/libs/form/components/NumberSlider.tsx

```diff
@@ -56,43 +77,73 @@ export function NumberSlider({
   );
 
   const handleSliderFocus = (event: React.FocusEvent<HTMLDivElement>) => {
-    setFocused(true);
+    setState((current) => ({ ...current, focused: true }));
     sliderProps?.onFocus?.(event);
   };
 
   const handleSliderBlur = (event: React.FocusEvent<HTMLDivElement>) => {
-    setFocused(false);
+    setState((current) => ({ ...current, focused: false }));
     sliderProps?.onBlur?.(event);
   };
 
   const handleInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
-    setFocused(true);
+    setState((current) => ({ ...current, focused: true }));
     numberProps?.onFocus?.(event);
   };
 
   const handleInputBlur = (event: React.FocusEvent<HTMLInputElement>) => {
-    setFocused(false);
+    setState((current) => ({ ...current, focused: false }));
     numberProps?.onBlur?.(event);
   };
 
   useEffect(() => {
-    if (!focused) {
-      setValue(value);
+    if (!state.focused) {
+      setState((current) => ({ ...current, _value: value }));
     }
-  }, [value, precision]);
+  }, [value, precision, state.focused]);
```

Comment:
This should pretty much cover the useEffect above, might wanna add the preset here too.